### PR TITLE
feat(clerk-js,types): Introduce logoLinkUrl prop in appearance

### PR DESCRIPTION
**Pull request by nikospapcom**
## Type of change

- [ ] 🐛 Bug fix
- [x] 🌟 New feature
- [ ] 🔨 Breaking change
- [ ] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

## Packages affected

- [x] `@clerk/clerk-js`
- [ ] `@clerk/clerk-react`
- [ ] `@clerk/nextjs`
- [ ] `@clerk/remix`
- [x] `@clerk/types`
- [ ] `@clerk/themes`
- [ ] `@clerk/localizations`
- [ ] `@clerk/clerk-expo`
- [ ] `@clerk/backend`
- [ ] `@clerk/clerk-sdk-node`
- [ ] `@clerk/shared`
- [ ] `@clerk/fastify`
- [ ] `@clerk/chrome-extension`
- [ ] `gatsby-plugin-clerk`
- [ ] `build/tooling/chore`

## Description

- [ ] `npm test` runs as expected.
- [ ] `npm run build` runs as expected.

### CODE REVIEW

**Comment by nikospapcom** [comment]
!snapshot

**Comment by clerk-cookie** [comment]
Hey @nikospapcom - the snapshot version command generated the following package versions:
| Package | Version |
| --- | --- |
| @clerk/backend | 0.23.8-snapshot.ab1d310 |
| @clerk/chrome-extension | 0.3.20-snapshot.ab1d310 |
| @clerk/clerk-js | 4.52.0-snapshot.ab1d310 |
| eslint-config-custom | 0.3.0 |
| @clerk/clerk-expo | 0.18.11-snapshot.ab1d310 |
| @clerk/fastify | 0.5.7-snapshot.ab1d310 |
| gatsby-plugin-clerk | 4.3.20-snapshot.ab1d310 |
| @clerk/localizations | 1.22.1-snapshot.ab1d310 |
| @clerk/nextjs | 4.21.14-snapshot.ab1d310 |
| @clerk/clerk-react | 4.21.2-snapshot.ab1d310 |
| @clerk/remix | 2.6.17-snapshot.ab1d310 |
| @clerk/clerk-sdk-node | 4.10.15-snapshot.ab1d310 |
| @clerk/shared | 0.19.1 |
| @clerk/themes | 1.7.5 |
| @clerk/types | 3.47.0-snapshot.ab1d310 |

Tip: use the snippet copy button below to quickly install the required packages.
```sh
# @clerk/backend
npm i @clerk/backend@0.23.8-snapshot.ab1d310
```
```sh
# @clerk/chrome-extension
npm i @clerk/chrome-extension@0.3.20-snapshot.ab1d310
```
```sh
# @clerk/clerk-js
npm i @clerk/clerk-js@4.52.0-snapshot.ab1d310
```
```sh
# eslint-config-custom
npm i eslint-config-custom@0.3.0
```
```sh
# @clerk/clerk-expo
npm i @clerk/clerk-expo@0.18.11-snapshot.ab1d310
```
```sh
# @clerk/fastify
npm i @clerk/fastify@0.5.7-snapshot.ab1d310
```
```sh
# gatsby-plugin-clerk
npm i gatsby-plugin-clerk@4.3.20-snapshot.ab1d310
```
```sh
# @clerk/localizations
npm i @clerk/localizations@1.22.1-snapshot.ab1d310
```
```sh
# @clerk/nextjs
npm i @clerk/nextjs@4.21.14-snapshot.ab1d310
```
```sh
# @clerk/clerk-react
npm i @clerk/clerk-react@4.21.2-snapshot.ab1d310
```
```sh
# @clerk/remix
npm i @clerk/remix@2.6.17-snapshot.ab1d310
```
```sh
# @clerk/clerk-sdk-node
npm i @clerk/clerk-sdk-node@4.10.15-snapshot.ab1d310
```
```sh
# @clerk/shared
npm i @clerk/shared@0.19.1
```
```sh
# @clerk/themes
npm i @clerk/themes@1.7.5
```
```sh
# @clerk/types
npm i @clerk/types@3.47.0-snapshot.ab1d310
```

File: packages/clerk-js/src/ui/elements/ApplicationLogo.tsx
Comment (desiprisg) [inline packages/clerk-js/src/ui/elements/ApplicationLogo.tsx:32]: Do we want to use this value that's set from the accounts portal on the components after all?

File: .changeset/wise-badgers-cover.md
Comment (nikosdouvlis) [inline .changeset/wise-badgers-cover.md:6]: 👋🏻 
What is the use case for this prop? Can we add more info in the description, please as this prop sounds a bit too specific to be in ClerkProvider

Comment (nikosdouvlis) [reply .changeset/wise-badgers-cover.md:6]: Is this only going to be used by Hosted Pages for now?

Comment (nikospapcom) [reply .changeset/wise-badgers-cover.md:6]: no @nikosdouvlis 
is someone use clerk-js will able to pass `afterLogoClickUrl` in order to change the `ApplicationLogo` url or otherwise if someone use hosted pages will able to change it throw dashboard

File: packages/clerk-js/src/ui/elements/ApplicationLogo.tsx
Comment (nikospapcom) [reply packages/clerk-js/src/ui/elements/ApplicationLogo.tsx:32]: Both @desiprisg, we want to change the ApplcationLogo from the accounts portal and components

File: .changeset/hungry-radios-protect.md
Comment (nikosdouvlis) [inline .changeset/hungry-radios-protect.md:5]: We probably need to rebase this PR?

File: .changeset/wise-badgers-cover.md
Comment (desiprisg) [reply .changeset/wise-badgers-cover.md:6]: I can also see that if someone has this set in the dashboard, it will also affect the components as we are falling back on it if a prop is not passed to `<ClerkProvider/>`. That's what I mentioned above.

Comment (desiprisg) [reply .changeset/wise-badgers-cover.md:6]: I though we were going to ignore this in `clerk-js` and just use the prop on accounts.

**Comment by nikospapcom** [comment]
!snapshot

**Comment by clerk-cookie** [comment]
Hey @nikospapcom - the snapshot version command generated the following package versions:
| Package | Version |
| --- | --- |
| @clerk/backend | 0.24.1-snapshot.27b5e0a |
| @clerk/chrome-extension | 0.3.22-snapshot.27b5e0a |
| @clerk/clerk-js | 4.53.0-snapshot.27b5e0a |
| eslint-config-custom | 0.3.0 |
| @clerk/clerk-expo | 0.18.13-snapshot.27b5e0a |
| @clerk/fastify | 0.5.8-snapshot.27b5e0a |
| gatsby-plugin-clerk | 4.3.21-snapshot.27b5e0a |
| @clerk/localizations | 1.22.2-snapshot.27b5e0a |
| @clerk/nextjs | 4.21.15-snapshot.27b5e0a |
| @clerk/clerk-react | 4.22.1-snapshot.27b5e0a |
| @clerk/remix | 2.7.1-snapshot.27b5e0a |
| @clerk/clerk-sdk-node | 4.10.16-snapshot.27b5e0a |
| @clerk/shared | 0.19.1 |
| @clerk/themes | 1.7.5 |
| @clerk/types | 3.47.0-snapshot.27b5e0a |

Tip: use the snippet copy button below to quickly install the required packages.
```sh
# @clerk/backend
npm i @clerk/backend@0.24.1-snapshot.27b5e0a
```
```sh
# @clerk/chrome-extension
npm i @clerk/chrome-extension@0.3.22-snapshot.27b5e0a
```
```sh
# @clerk/clerk-js
npm i @clerk/clerk-js@4.53.0-snapshot.27b5e0a
```
```sh
# eslint-config-custom
npm i eslint-config-custom@0.3.0
```
```sh
# @clerk/clerk-expo
npm i @clerk/clerk-expo@0.18.13-snapshot.27b5e0a
```
```sh
# @clerk/fastify
npm i @clerk/fastify@0.5.8-snapshot.27b5e0a
```
```sh
# gatsby-plugin-clerk
npm i gatsby-plugin-clerk@4.3.21-snapshot.27b5e0a
```
```sh
# @clerk/localizations
npm i @clerk/localizations@1.22.2-snapshot.27b5e0a
```
```sh
# @clerk/nextjs
npm i @clerk/nextjs@4.21.15-snapshot.27b5e0a
```
```sh
# @clerk/clerk-react
npm i @clerk/clerk-react@4.22.1-snapshot.27b5e0a
```
```sh
# @clerk/remix
npm i @clerk/remix@2.7.1-snapshot.27b5e0a
```
```sh
# @clerk/clerk-sdk-node
npm i @clerk/clerk-sdk-node@4.10.16-snapshot.27b5e0a
```
```sh
# @clerk/shared
npm i @clerk/shared@0.19.1
```
```sh
# @clerk/themes
npm i @clerk/themes@1.7.5
```
```sh
# @clerk/types
npm i @clerk/types@3.47.0-snapshot.27b5e0a
```

**Comment by nikospapcom** [comment]
!snapshot

**Comment by clerk-cookie** [comment]
Hey @nikospapcom - the snapshot version command generated the following package versions:
| Package | Version |
| --- | --- |
| @clerk/backend | 0.24.1-snapshot.4993ef8 |
| @clerk/chrome-extension | 0.3.22-snapshot.4993ef8 |
| @clerk/clerk-js | 4.53.0-snapshot.4993ef8 |
| eslint-config-custom | 0.3.0 |
| @clerk/clerk-expo | 0.18.13-snapshot.4993ef8 |
| @clerk/fastify | 0.5.8-snapshot.4993ef8 |
| gatsby-plugin-clerk | 4.3.21-snapshot.4993ef8 |
| @clerk/localizations | 1.22.2-snapshot.4993ef8 |
| @clerk/nextjs | 4.21.15-snapshot.4993ef8 |
| @clerk/clerk-react | 4.22.1-snapshot.4993ef8 |
| @clerk/remix | 2.7.1-snapshot.4993ef8 |
| @clerk/clerk-sdk-node | 4.10.16-snapshot.4993ef8 |
| @clerk/shared | 0.19.1 |
| @clerk/themes | 1.7.5 |
| @clerk/types | 3.47.0-snapshot.4993ef8 |

Tip: use the snippet copy button below to quickly install the required packages.
```sh
# @clerk/backend
npm i @clerk/backend@0.24.1-snapshot.4993ef8
```
```sh
# @clerk/chrome-extension
npm i @clerk/chrome-extension@0.3.22-snapshot.4993ef8
```
```sh
# @clerk/clerk-js
npm i @clerk/clerk-js@4.53.0-snapshot.4993ef8
```
```sh
# eslint-config-custom
npm i eslint-config-custom@0.3.0
```
```sh
# @clerk/clerk-expo
npm i @clerk/clerk-expo@0.18.13-snapshot.4993ef8
```
```sh
# @clerk/fastify
npm i @clerk/fastify@0.5.8-snapshot.4993ef8
```
```sh
# gatsby-plugin-clerk
npm i gatsby-plugin-clerk@4.3.21-snapshot.4993ef8
```
```sh
# @clerk/localizations
npm i @clerk/localizations@1.22.2-snapshot.4993ef8
```
```sh
# @clerk/nextjs
npm i @clerk/nextjs@4.21.15-snapshot.4993ef8
```
```sh
# @clerk/clerk-react
npm i @clerk/clerk-react@4.22.1-snapshot.4993ef8
```
```sh
# @clerk/remix
npm i @clerk/remix@2.7.1-snapshot.4993ef8
```
```sh
# @clerk/clerk-sdk-node
npm i @clerk/clerk-sdk-node@4.10.16-snapshot.4993ef8
```
```sh
# @clerk/shared
npm i @clerk/shared@0.19.1
```
```sh
# @clerk/themes
npm i @clerk/themes@1.7.5
```
```sh
# @clerk/types
npm i @clerk/types@3.47.0-snapshot.4993ef8
```

File: .changeset/wise-badgers-cover.md
Comment (yourtallness) [inline .changeset/wise-badgers-cover.md:8]: 🤓
> otherwise logo url will be set to `homeUrl`

File: packages/clerk-js/src/ui/customizables/__tests__/parseAppearance.test.tsx
Comment (yourtallness) [inline packages/clerk-js/src/ui/customizables/__tests__/parseAppearance.test.tsx:236]: @nikosdouvlis @desiprisg Not related to this PR, but it would be nice to have some actual test URLs here.

**Review by yourtallness — APPROVED** [review]
@nikospapcom other than a phrasing correction, #lgtm

But please also wait for an approval from the JS team.

File: .changeset/wise-badgers-cover.md
Comment (nikosdouvlis) [inline .changeset/wise-badgers-cover.md:7]: 💯

File: packages/clerk-js/src/ui/customizables/__tests__/parseAppearance.test.tsx
Comment (nikosdouvlis) [reply packages/clerk-js/src/ui/customizables/__tests__/parseAppearance.test.tsx:236]: Yup makes sense - adding a ticket

File: packages/types/src/appearance.ts
Comment (nikosdouvlis) [inline packages/types/src/appearance.ts:571]:
```suggestion
  /**
   * Controls where the browser will redirect to after the user clicks the application logo, usually found in the SignIn and SignUp components. If a URL is provided, it will be used as the `href` of the link.
   * @default Clicking clicking the application logo will redirect to Home Url, as set in the Clerk Dashboard 
   */
```

**Review by nikosdouvlis — APPROVED** [review]
A minor comment, approving to unblock :)
Good job @nikospapcom

File: .changeset/wise-badgers-cover.md
Comment (yourtallness) [inline .changeset/wise-badgers-cover.md:7]: 🤓 "the target of the logo"

**Comment by nikospapcom** [comment]
!snapshot

**Comment by clerk-cookie** [comment]
Hey @nikospapcom - the snapshot version command generated the following package versions:
| Package | Version |
| --- | --- |
| @clerk/backend | 0.27.0-snapshot.8b4eb49 |
| @clerk/chrome-extension | 0.3.26-snapshot.8b4eb49 |
| @clerk/clerk-js | 4.55.0-snapshot.8b4eb49 |
| eslint-config-custom | 0.3.0 |
| @clerk/clerk-expo | 0.18.17-snapshot.8b4eb49 |
| @clerk/fastify | 0.6.3-snapshot.8b4eb49 |
| gatsby-plugin-clerk | 4.4.4-snapshot.8b4eb49 |
| @clerk/localizations | 1.24.1-snapshot.8b4eb49 |
| @clerk/nextjs | 4.23.2-snapshot.8b4eb49 |
| @clerk/clerk-react | 4.23.2-snapshot.8b4eb49 |
| @clerk/remix | 2.9.1-snapshot.8b4eb49 |
| @clerk/clerk-sdk-node | 4.12.2-snapshot.8b4eb49 |
| @clerk/shared | 0.21.0-snapshot.8b4eb49 |
| @clerk/themes | 1.7.5 |
| @clerk/types | 3.49.0-snapshot.8b4eb49 |

Tip: use the snippet copy button below to quickly install the required packages.
```sh
# @clerk/backend
npm i @clerk/backend@0.27.0-snapshot.8b4eb49
```
```sh
# @clerk/chrome-extension
npm i @clerk/chrome-extension@0.3.26-snapshot.8b4eb49
```
```sh
# @clerk/clerk-js
npm i @clerk/clerk-js@4.55.0-snapshot.8b4eb49
```
```sh
# eslint-config-custom
npm i eslint-config-custom@0.3.0
```
```sh
# @clerk/clerk-expo
npm i @clerk/clerk-expo@0.18.17-snapshot.8b4eb49
```
```sh
# @clerk/fastify
npm i @clerk/fastify@0.6.3-snapshot.8b4eb49
```
```sh
# gatsby-plugin-clerk
npm i gatsby-plugin-clerk@4.4.4-snapshot.8b4eb49
```
```sh
# @clerk/localizations
npm i @clerk/localizations@1.24.1-snapshot.8b4eb49
```
```sh
# @clerk/nextjs
npm i @clerk/nextjs@4.23.2-snapshot.8b4eb49
```
```sh
# @clerk/clerk-react
npm i @clerk/clerk-react@4.23.2-snapshot.8b4eb49
```
```sh
# @clerk/remix
npm i @clerk/remix@2.9.1-snapshot.8b4eb49
```
```sh
# @clerk/clerk-sdk-node
npm i @clerk/clerk-sdk-node@4.12.2-snapshot.8b4eb49
```
```sh
# @clerk/shared
npm i @clerk/shared@0.21.0-snapshot.8b4eb49
```
```sh
# @clerk/themes
npm i @clerk/themes@1.7.5
```
```sh
# @clerk/types
npm i @clerk/types@3.49.0-snapshot.8b4eb49
```

**Review by SokratisVidros — APPROVED** [review]

**Review by nikosdouvlis — APPROVED** [review]

**Comment by clerk-cookie** [comment]
This PR has been automatically locked since there has not been any recent activity after it was closed. Please open a new issue for related bugs.

**Pull request merged**
